### PR TITLE
chore: update all the things

### PR DESCRIPTION
**Pull request by esacteksab**
- Updates all the things
- Adds Poetry
- Gets rid of pipenv
- Gets rid of requirements.txt
- Tests pass locally

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+multidict = "4.5.2"
+pycares = "4.3.0"
+pycparser = "2.19"
+slackclient = "2.9.4"
+werkzeug = "2.2.3"
+yarl = "1.8.2"
+itsdangerous = "^2.1.2"
+flask = "^2.2.3"
+
+
```
Comment (rpdelaney) [inline pyproject.toml:30]:
```suggestion
slackclient = "2.9.4"
flask = "^2.2.3"

```
This is enough for poetry to do the rest

**Review by rpdelaney — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+[tool.poetry]
+name = "affinity-groups-slackbot"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
+readme = "README.md"
+packages = [{include = "affinity_groups_slackbot"}]
+
+[tool.poetry.dependencies]
+python = "^3.11"
```
Comment (rpdelaney) [inline pyproject.toml:10]:
```suggestion
python = "^3.10"
```
I think this will fix CI

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+[tool.poetry]
+name = "affinity-groups-slackbot"
+description = ""
+authors = ["Your Name <you@example.com>"]
```
Comment (rpdelaney) [inline pyproject.toml:4]:
```suggestion
authors = ["TrussWorks, Inc. <why@truss.works>"]
```

**Comment by esacteksab** [comment]
This should also close #7, #8 and #9!

**Review by rpdelaney — APPROVED** [review]
We'll get renovate in separately...

**Comment by esacteksab** [comment]
> We'll get renovate in separately...

Sorry! Can't believe I forgot that!!

**Pull request merged**